packages/data-stores/src/launch/actions.ts

```diff
@@ -124,14 +143,35 @@ export const hideModalTitle = () =>
 		type: 'HIDE_MODAL_TITLE',
 	} as const );
 
+export const enablePersistentSuccessView = () =>
+	( {
+		type: 'ENABLE_SUCCESS_VIEW',
+	} as const );
+
+export const disablePersistentSuccessView = () =>
+	( {
+		type: 'DISABLE_SUCCESS_VIEW',
+	} as const );
+
+// TODO: I feel that there is a better way to type this
+type SetPlanActionParams = () =>
+	| {
+			readonly type: 'SET_PLAN_BILLING_PERIOD';
+			readonly billingPeriod: Plans.PlanBillingPeriod;
+	  }
+	| {
+			readonly type: 'SET_PLAN_PRODUCT_ID';
+			readonly planProductId: number | undefined;
+	  };
+
```

Comment:
@sirreal Hi Jon! I couldn't figure out a way to type a generator action creator. Do you have a more elegant solution? This one isn't quite maintainable.

Reply:
How about swapping `ReturnType` below for a new mapped type helper:

```ts
type ReturnOrGeneratorYieldUnion<T extends (...args: any) => any> = T extends (
	...args: any
) => infer Return
	? (Return extends Generator<infer T, infer U, any>
		? T | U
		: Return)
	: never;

export type LuanchAction = ReturnOrGeneratorYieldUnion<
  | typeof setPlanProductId
  | typeof setSiteTitle /* … */;
```

I think you'll get the result you're looking for. [Here's a playground link](https://www.typescriptlang.org/play?#code/C4TwDgpgBAShwFcBOA7A8kg4hFEkENgB7JATQEsIAbAEwB4AVKCAD2BxoGcoAKAOgH4kAc04AuKPhQgAlFAC8APknTl8qE1bsUXXgCgAkAL5DREqSD1ylUcigBmeWPGQpDAfl5xEqZmw7c2LgExEh0do5IGgA0tg5OAKqxFoqGBp5MAD5QCWkS3q4yhhK4AG54ANx6AMZEKJzAUPZERAq81so8AN5Q4lAARJz9UAC+kty19cAyVXqgkDltBagYQXiEJBTU9PMQRPa98AAKVFJHSEQ0CNXAAJI0UNm7+00tirOsYCSNkw2HwCczhcrjd7m17AgUDdyHUAFT-QEoc6Xa53Gg8QxgU5I4Go+4lBAAWwARk5spCaBB7HYIDQrFAuoZfo1yJwAGJICDQdTAJAIaD4bjElpUCBSWYGcgHHgAQlZHK5ckZBgMzKgxPIVCodmERzwMIe6n6AEEAHKmhLGgAyVtI-QlBhAlFoUAA+q7OPAAEKa7UoXX6y48DVanV6pAGmaGEZ6QycnwoBlpXYSfoAZQAogxXUcrWaczA0AARBIAYWztyL-WiaSxQJRoLpBjGgqgzKqI1mEKhwBhifdnuAPtD-vDBuDvrDgZoEgaEf9SrjLl8yoMKYGmezufzXtuNtupswOYzMFuxeraRDfoDEcu0fGbbqDQ7saAA).

Reply:
Thank you so much! We could add this type to wp.data pacakge. Since it will be needed whenever some needs a generator action.

Reply:
I'm undecided between that above and this simpler solution:

```ts
export type LaunchAction =
	| ReturnType<
			| typeof setSiteTitle
			.
			.
			| typeof hideModalTitle
	  >
	| ReturnType< ReturnType< typeof setPlanProductId >[ 'next' ] >[ 'value' ]; // <--- this
```

Reply:
That's simpler here for a one-off case. I'd like to see a library-level solution for this that we could life to Gutenberg when it's ready like you suggest.